views/js/runner/plugins/navigation/next/nextItemWarning.js

```diff
@@ -0,0 +1,155 @@
+                var checkboxParams = null;
+
+                testRunner.trigger('disablenav');
+
+                return new Promise(function(resolve, reject) {
+                    // Load testStore checkbox value (async)
+                    testStore.getStore(self.getName()).then(function(store) {
+                        store.getItem('dontShowNextItemWarning').then(function(checkboxValue) {
+
+                            function cancel() {
+                                testRunner.trigger('enablenav');
+                                reject();
+                            }
+
+                            // Show the warning unless user has turned it off:
+                            if (checkboxValue !== true) {
+                                // Define checkbox only if enabled by config:
+                                if (testConfig.enableNextItemWarningCheckbox) {
+                                    checkboxParams = {
+                                        checked: checkboxValue,
+                                        submitChecked: function() {
+                                            store.setItem('dontShowNextItemWarning', true);
+                                        },
+                                        submitUnchecked: function() {
+                                            store.setItem('dontShowNextItemWarning', false);
+                                        },
+                                    };
+                                }
+                                // show special dialog:
+                                dialogConfirmNext(
+                                    __('Go to the next item?'),
+                                    getCustomNextMessage(action),
+                                    resolve, // if the test taker accepts
+                                    cancel,  // if he refuses
+                                    checkboxParams
+                                );
+                            }
+                            else {
+                                resolve();
+                            }
+                        });
+                    });
+                });
+            }
+
+            // Attach this plugin to 'next' & 'skip' events
+            testRunner
+                .on('init', function() {
+                    // Clear the stored checkbox value before each test:
+                    testStore.getStore(self.getName()).then(function(store) {
+                        store.setItem('dontShowNextItemWarning', null);
+                    });
+                })
+                .before('move skip', function(e, type, scope) {
+                    var context = testRunner.getTestContext();
+                    if (context.isLinear) {
+                        // Do nothing if nextSection warning imminent:
+                        if (scope === 'section' && context.options.nextSectionWarning) {
+                            return Promise.resolve();
```

Comment:
this is not necessary to return a Promise if you don't need to block the event. You should handle only the cases where you'll block the event